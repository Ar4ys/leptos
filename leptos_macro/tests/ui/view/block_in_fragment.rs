use leptos::*;

struct A;

fn main() {
    view! {
        <>{A}</>
    };
}
