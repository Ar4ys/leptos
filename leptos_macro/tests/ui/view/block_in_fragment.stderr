error[E0277]: expected a `Fn()` closure, found `A`
 --> tests/ui/view/block_in_fragment.rs:7:11
  |
6 | /     view! {
7 | |         <>{A}</>
  | |           ^-^
  | |           ||
  | |           |this tail expression is of type `A`
  | |           expected an `Fn()` closure, found `A`
8 | |     };
  | |_____- required by a bound introduced by this call
  |
  = help: the trait `std::ops::Fn<()>` is not implemented for `A`
  = note: wrap the `A` in a closure with no arguments: `|| { /* code */ }`
  = help: the following other types implement trait `IntoView`:
            bool
            char
            isize
            i8
            i16
            i32
            i64
            i128
          and $N others
  = note: required for `A` to implement `IntoView`
help: you might have meant to create the closure instead of a block
  |
7 |         <>|_| {A}</>
  |           +++
